spring-core/src/main/java/org/springframework/core/CoroutinesUtils.java

```diff
@@ -71,7 +75,9 @@
 	public static Publisher<?> invokeSuspendingFunction(Method method, Object target, Object... args) {
 		KFunction<?> function = Objects.requireNonNull(ReflectJvmMapping.getKotlinFunction(method));
 		KClassifier classifier = function.getReturnType().getClassifier();
-		Mono<Object> mono = MonoKt.mono(Dispatchers.getUnconfined(), (scope, continuation) ->
+		Continuation<?> cont = (Continuation<?>) args[args.length - 1];
```

Comment:
I am not super confortable doing this kind of manual handling of the continuation, so I have asked guidance to Kotlin team to see if there is a better way, waiting there feedback.

Reply:
Feedback from Kotlin team:

> What puzzles me is why there is `Job` in the context.
> When you have one, it means that the lifecycle of the result is just incorrect — it is controlled solely by mono, but not by the Job as the caller might’ve expected.
> It’s hard to tell if it is exactly wrong without knowing why Job is there in the first place

Potentially related to spring-projects/spring-data-commons#2532 cc @mp911de.

Reply:
It is indeed the same for WebFlux. I will provide your feedback to the Kotlin team, may be indeed ok for this specific use case.

Reply:
@qwwdfsad Could you please check @ks-yim previous comment, especially this quote below, to help us analyzing if removing `Job.Key` from the `CoroutineContext` is ok for this very specific use case where we intercept the suspending invocation to plug our Reactive transaction management and later invoke `AwaitKt.awaitSingleOrNull` with the original continuation:
> I understand the concerns of Kotlin team worrying that the coroutine's lifecycle is no longer be managed by the caller's Job context, but I thought it's okay to do this for this specific use case since the lifecycle control of the returned Mono goes back into the caller's Job context anyways in the subsequent call for awaitSingleOrNull in [this line](https://github.com/spring-projects/spring-framework/blob/main/spring-tx/src/main/java/org/springframework/transaction/interceptor/TransactionAspectSupport.java#L370-L372).

FWIW our tests are green with this implementation proposed in [my refined commit](https://github.com/sdeleuze/spring-framework/commit/gh-27308).

Reply:
(Another guy from the coroutines project here) You are doing very fancy things here! It did take me a while to understand them. If the `CompletableFuture` gets cancelled, it will cancel `awaitSingleOrNull`; if `awaitSingleOrNull` gets cancelled, it will cancel the `mono`; if `mono` completes with an error or a value, it will return that via `awaitSingleOrNull`. So, in effect, you're getting the same job control with extra steps.

The only issue I see is coroutine code that explicitly does something with the `Job`, like `currentCoroutineContext()[Job]!!.invokeOnCompletion { println("ok") }`. If you intercept such code in a `mono`, then, naturally, the `Job` used will be the one created by `mono`, not the `Job` from `future`. I don't see a way around this, though.